### PR TITLE
fix: Avoid duplicate rule evaluation for rules with multiple definitions

### PR DESCRIPTION
**Pull request by liamg**

### CODE REVIEW

File: pkg/scanners/helm/parser/parser.go [hunk 1]
```diff
@@ -232,7 +232,6 @@ func (p *Parser) writeBuildFiles(tempFs string) error {
 		if err != nil {
 			return err
 		}
-		fmt.Printf("path: %s, rootPath: %s\n", path, p.rootPath)
```
Comment (owenrumney) [inline pkg/scanners/helm/parser/parser.go:235]: I see you're cleaning up after me.....

File: pkg/scanners/helm/test/scanner_test.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func Test_helm_scanner_with_archive(t *testing.T) {
 		require.NotNil(t, results)
 
 		failed := results.GetFailed()
-		assert.Equal(t, 26, len(failed))
+		assert.Equal(t, 25, len(failed))
```
Comment (owenrumney) [inline pkg/scanners/helm/test/scanner_test.go:50]: how come we lost one?

**Review by owenrumney — APPROVED** [review]

File: pkg/scanners/helm/parser/parser.go (hunk 1 above)
Comment (liamg) [reply pkg/scanners/helm/parser/parser.go:235]: :broom: :broom:

Yeah, test failed as i fixed the rego getting run twice so the number was different, so noticed the output

File: pkg/scanners/helm/test/scanner_test.go (hunk 2 above)
Comment (liamg) [reply pkg/scanners/helm/test/scanner_test.go:50]: i fixed the rego getting run twice so the number was different

Comment (liamg) [reply pkg/scanners/helm/test/scanner_test.go:50]: i had a bug where if you had 2 rules called `deny` it would run them both twice

**Pull request merged**